packages/js/product-editor/src/components/attribute-control/new-attribute-modal.tsx

```
@@ -239,6 +246,62 @@ export const NewAttributeModal: React.FC< NewAttributeModalProps > = ( {
 					// eslint-disable-next-line @typescript-eslint/no-explicit-any
 					setValue: ( name: string, value: any ) => void;
 				} ) => {
+					function getAttributeOnChange( index: number ) {
+						return function handleAttributeChange(
+							value?:
+								| Omit<
+										ProductAttribute,
+										'position' | 'visible' | 'variation'
+								  >
+								| string
+						) {
+							if (
+								termsAutoselection &&
+								value &&
+								! ( typeof value === 'string' )
+							) {
+								resolveSelect(
+									EXPERIMENTAL_PRODUCT_ATTRIBUTE_TERMS_STORE_NAME
+								)
+									.getProductAttributeTerms<
+										ProductAttributeTerm[]
+									>( {
+										// Send search parameter as empty to avoid a second
+										// request when focusing the attribute-term-input-field
+										// which perform the same request to get all the terms
+										search: '',
+										attribute_id: value.id,
+									} )
+									.then( ( terms ) => {
+										const selectedAttribute =
+											getProductAttributeObject(
+												value
+											) as EnhancedProductAttribute;
+										if ( termsAutoselection === 'all' ) {
+											selectedAttribute.terms = terms;
+										} else if ( terms.length > 0 ) {
+											selectedAttribute.terms = [
+												terms[ 0 ],
+											];
+										}
+										setValue( 'attributes[' + index + ']', {
+											...selectedAttribute,
+										} );
+										focusValueField( index );
+									} )
+									.catch( () => {} );
```

I think this would actually suppress an error, making it invisible. We should either remove this or add some error handling here

---

done -> https://github.com/woocommerce/woocommerce/pull/40004/commits/9313f2e13cf80ee7a720744aa991aa0652c7730d